tests/functional/test_investment_pathway.py

```diff
@@ -0,0 +1,476 @@
+# Copyright (c) 2024, RTE (https://www.rte-france.com)
+#
+# See AUTHORS.txt
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# SPDX-License-Identifier: MPL-2.0
+#
+# This file is part of the Antares project.
+
+import pytest
+from anytree import Node as TreeNode
+
+from andromede.expression import literal, var
+from andromede.expression.indexing_structure import IndexingStructure
+from andromede.libs.standard import (
+    DEMAND_MODEL,
+    GENERATOR_MODEL,
+    NODE_WITH_SPILL_AND_ENS,
+    THERMAL_CANDIDATE_WITH_ALREADY_INSTALLED_CAPA,
+)
+from andromede.model.common import ProblemContext
+from andromede.model.constraint import Constraint
+from andromede.model.model import model
+from andromede.model.variable import float_variable
+from andromede.simulation import (
+    BendersSolution,
+    TimeBlock,
+    build_benders_decomposed_problem,
+)
+from andromede.simulation.decision_tree import (
+    DecisionTreeNode,
+    InterDecisionTimeScenarioConfig,
+)
+from andromede.study.data import ConstantData, DataBase, TreeData
+from andromede.study.network import Component, Network, Node, PortRef, create_component
+
+
+@pytest.fixture
+def generator() -> Component:
+    generator = create_component(
+        model=GENERATOR_MODEL,
+        id="BASE",
+    )
+    return generator
+
+
+@pytest.fixture
+def candidate() -> Component:
+    candidate = create_component(
+        model=THERMAL_CANDIDATE_WITH_ALREADY_INSTALLED_CAPA, id="CAND"
+    )
+    return candidate
+
+
+@pytest.fixture
+def demand() -> Component:
+    demand = create_component(model=DEMAND_MODEL, id="D")
+    return demand
+
+
+@pytest.fixture
+def node() -> Node:
+    node = Node(model=NODE_WITH_SPILL_AND_ENS, id="N")
+    return node
+
+
+def test_investment_pathway_on_sequential_nodes(
```

Comment:
Maybe try to refactorize this test a bit so that we see more clearly the general structure : generate database, create coupling model / network, replicate on nodes (almost duplicated code on parent and child), decision tree building, ...

Reply:
Also write a comment that describes the test and so that we understand the expected solution at glance